rosbag2_cpp/test/rosbag2_cpp/test_sequential_writer.cpp

```diff
@@ -265,41 +265,6 @@ TEST_F(SequentialWriterTest, writer_splits_when_storage_bagfile_size_gt_max_bagf
   }
 }
 
-TEST_F(SequentialWriterTest, only_write_after_cache_is_full) {
```

Comment:
why is that test deleted?

Reply:
The cache semantic changed - we no longer want to only start writing when the cache is full (we want to keep writing at all times if there is anything in buffers to optimize I/O). Hence the test doesn't make sense.

However, what makes perfect sense is to replace the test with some new tests for cache!

Reply:
Added 2 tests